modules/db/src/main/java/org/mule/module/db/internal/config/domain/query/ParameterizedQueryTemplateFactoryBean.java

```diff
@@ -54,7 +54,11 @@ public QueryTemplate getObject() throws Exception
         {
             QueryParam param = findOverriddenParam(templateParam.getName(), queryParams);
 
-            if (param == null)
+            if (param == null && ((DefaultInputQueryParam) templateParam).isDbInParam())
```

Comment:
Because If It's a literal or a MEL expression, Validation isn't necessary (They doesn't refer to a db:in-param config).